packages/participants-svc/src/domain/participant_agg.ts

```diff
@@ -917,34 +916,182 @@ export class ParticipantAggregate {
+            currencyCode: accountChangeRequest.currencyCode,
+            creditBalance: null,
+            debitBalance: null,
+            balance: null,
+            externalBankAccountId: accountChangeRequest.externalBankAccountId,
+            externalBankAccountName: accountChangeRequest.externalBankAccountName,
+            createdBy: secCtx.username!,
+            createdDate: Date.now(),
+            approved: false,
+            approvedBy: null,
+            approvedDate: null
+        });
+        existing.changeLog.push({
+            changeType: ParticipantChangeTypes.CHANGE_ACCOUNT,
+            user: secCtx.username!,
+            timestamp: Date.now(),
+            notes: null,
+        });
+
+        const updateSuccess = await this._repo.store(existing);
+        if (!updateSuccess) {
+            const err = new CouldNotStoreParticipant(
+                "Could not update participant on addParticipantAccount"
             );
-        } catch (err) {
             this._logger.error(err);
-            if (err instanceof UnauthorizedError) throw err;
+            throw err;
+        }
+
+        this._logger.info(
+            `Successfully created account to Participant with ID: '${participantId}'`
+        );
+
+        await this._auditClient.audit(
+            AuditedActionNames.PARTICIPANT_ACCOUNT_CHANGE_REQUEST_CREATED,
+            true,
+            this._getAuditSecCtx(secCtx),
+            [{ key: "participantId", value: participantId }]
+        );
+
+        return accountChangeRequest.id;
+    }
+
+    async approveParticipantAccount(
+        secCtx: CallSecurityContext,
+        participantId: string,
+        accountChangeRequestId: string
+    ): Promise<string> {
+        this._enforcePrivilege(secCtx, ParticipantPrivilegeNames.APPROVE_ACCOUNTS_CHANGE_REQUEST);
 
-            throw new UnableToCreateAccountUpstream(
-                `'${existing.name}' account '${account.type}' failed upstream.`
+        if (!participantId)
+            throw new InvalidParticipantError("[id] cannot be empty");
+
+        const existing: IParticipant | null = await this._repo.fetchWhereId(
+            participantId
+        );
+        if (!existing)
+            throw new ParticipantNotFoundError(
+                `Participant with ID: '${participantId}' not found.`
+            );
+        // if (!existing.isActive) throw new ParticipantNotActive("Participant is not active.");
+
+        const accountChangeRequest = existing.participantAccountsChangeRequest.find(
+            (value: IParticipantAccountsChangeRequest) => value.id === accountChangeRequestId
+        );
+        if (!accountChangeRequest) {
+            throw new AccountNotFoundError(
+                `Cannot find a participant's account change request with id: ${accountChangeRequestId}`
+            );
+        }
+        if (accountChangeRequest.approved) {
+            throw new AccountNotFoundError(
```

Comment:
This should not be an AccountNotFoundError

Reply:
Used proper error type.